### DISCUSSION TITLE
Keep log of project breaking proposals after closing them

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#10086] [General Discussions]
A few times now i've seen user proposals closed with:

"Closing, as changing this now would break compatibility with existing projects", and the proposal is closed.

Closing these proposals means they will not be available at the big "window of opportunity" pre-5.0 round table meeting when it happens.

Leaving them open means they bloat the proposal stack.

PROPOSAL: There needs to be some log of these big suggestions that "break existing projects" so they can be given a chance to come through, be discussed at the big meeting when it happens, and not lost in the void.

**Comment by AThousandShips** [comment]
> A few times now i've seen user proposals closed with:

To make a good start for this, can you please link some of these? I don't recall any being closed *by maintainers* like that

Proposals shouldn't be, and in my experience *aren't* being, closed *just* because the proposed change breaks compatibility, they should (and are) marked as breaking compatibility and are marked for 5.0

That's assuming they are viable and good proposals otherwise

**Reply by shak2** [reply]
Example (looking at the date it is Oct 2023 so I don't know if this is still policy):

[https://github.com/godotengine/godot-proposals/issues/5169](https://github.com/godotengine/godot-proposals/issues/5169)

**Reply by AThousandShips** [reply]
That wasn't closed *just* for that reason, see the comment there, along with several other arguments. So that isn't really an example of what you're talking about. Do you have any other cases?

**Reply by Zireael07** [reply]
https://github.com/godotengine/godot-proposals/issues?q=is%3Aissue+is%3Aclosed+break+compatibility+ returns over 170 hits

**Reply by AThousandShips** [reply]
And many are closed as completed 🙃 and many others are duplicates, could you point to one closed for the reason of compatibility by a maintainer? And not a duplicate, completed, or abandoned by the author? (Narrowing it by adding the archived label already removes almost 100 of those)

Point being: can anyone show that the issue raised exists and that there's a need for this? Because it isn't policy, that's precisely why we have the 5.0 milestone in the first place

**Reply by Zireael07** [reply]
Those that show as closed as completed likely predate the introduction of ability to close as duplicate/invalid

That said the 5.0 milestone is probably a good solution

**Comment by AThousandShips** [comment]
But regardless the solution isn't IMO to track these supposed cases, but to use the 5.0 milestone and not closing them, which is as far as I know policy and how it's generally done

However if there's some other reason a proposal is closed, or if the original author closes it themselves, there's no reason to track them

If you find some examples of this happening then please link them and they should probably be considered differently, but I haven't found any like that

**Reply by shak2** [reply]
I did not know about the 5.0 milestoning/tagging which is as good as my log solution. If I find any new instances I will make an update or post, but for now I rest my case.
